### ISSUE TITLE
Live multipack DataLoader ignores training.batch_size, silently packing every bin to max_length only

### ISSUE THREAD

**Issue by AmirF194** [MakazhanAlpamys/Soup#561]
### Description

The live multipack `DataLoader` wiring ignores `training.batch_size` for packing density. `get_train_dataloader` in `make_multipack_trainer_class` (`src/soup_cli/utils/multipack_trainer.py`) builds the FFD sampler with `batch_max_len=int(max_seq)`, capping every packed bin at the raw sequence length instead of `batch_size * max_seq_length`. `src/soup_cli/utils/multipack.py`'s own `build_multipack_sampler_for_lengths` already implements the correct branch for this mode (`batch_max_len = batch_size * max_seq_length` when `real_batches=False`), but the live trainer wiring does not call it and never multiplies `batch_size` in.

The result: every multipack run gets the packing density of `batch_size=1` regardless of what `training.batch_size` is set to. This is silent: nothing crashes, the loss curve still looks healthy, only throughput on the packed path is much lower than the docs describe (`docs/peft-and-efficiency.md` calls multipack "Soup's largest single throughput win on chat fine-tuning").

Root cause: `6fdf7e2` (v0.40.4 Part B, "Closes #65") flipped the sampler from `real_batches=True` to `real_batches=False` to fix a `Sampler[int]` vs `list[list[int]]` shape mismatch, but left `batch_max_len=int(max_seq)` untouched, carrying the `real_batches=True`-era value into the `real_batches=False` branch where the semantics differ. Nothing has revisited that line since.

### Steps to reproduce

Found by reading the live wiring, not via `soup train`; reproduced standalone against the real source (no GPU/torch needed, `multipack_trainer.py` is pure Python until `get_train_dataloader` actually runs):

1. `attach_multipack_state(trainer, lengths=[...], max_seq_len=128, batch_size=4, seed=42)`
2. `dl = trainer.get_train_dataloader()`
3. `dl.batch_sampler._batch_max_len` is `128`, not `512` (`128 * 4`).

### Config (soup.yaml)

```yaml
training:
  multipack: true
  batch_size: 4
data:
  max_length: 128
```

### Error output

None. This is a silent throughput regression, not a crash.

### Environment

Not reproduced through the CLI (`soup doctor` not run) - found and confirmed by reading `src/soup_cli/utils/multipack_trainer.py` and `src/soup_cli/utils/multipack_sampler.py` against current `main` (`6c13c44`), and by exercising `get_train_dataloader` directly with a stand-in trainer object.

**Comment by MakazhanAlpamys** [comment]
Closed by #562 from @AmirF194. The flat multipack DataLoader now scales its FFD bin cap by training.batch_size as the helper and docs promise; reverting to bare max_seq is caught by the regression test.
